cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/auth/OAuth2TokenProvider.java

```diff
@@ -7,4 +7,10 @@ public interface OAuth2TokenProvider {
     OAuth2Token getToken() throws IOException;
 
     OAuth2Token refreshToken() throws IOException;
+
+    String getKeyForTokenIfParameter();
+
+    default boolean useAuthorizationHeader() {
+        return getKeyForTokenIfParameter() != null;
```

Comment:
Is this condition correct? The way I read it is that I should use the Authorization Header method if I am setup to use the token as a parameter.

If I don't have the token as a parameter, then I should not use the authorization header method.